osu.Game/Overlays/Profile/Sections/Recent/DrawableRecentActivity.cs

```diff
@@ -119,6 +121,105 @@ private Drawable createIcon()
                         Height = 18
                     };
 
+                case RecentActivityType.UserSupportAgain:
+                    return new SupporterIcon(SupporterType.SupportAgain)
```

Comment:
I really dislike how this code is structured. It's overly terse for what it needs to be. To demonstrate, here's a textual description of what this code is doing:

- We start off with a `RecentActivityType`
- This recent activity type is then split into two types of icon (either a supporter icon, or beatmap icon), for not much reason
- Both types of icon are also associated with their own enum, which maps 1:1 to `RecentActivityType` anyways, introducing another 1:1 mapping step that doesn't bring much at all
- And in the icons themselves, for whatever reason the type of icon and colour of icon are separated into two methods, making things even more difficult to follow

At the same time, all of the enum cases here are specifying `RelativeSizeAxes`, `Height` and `FillMode` and `Margin` over and over with the exact same values for not much reason at all.

I propose getting rid of `SupporterIcon` and `BeatmapIcon`, getting rid of their accompanying enumeration types, and removing the `getIcon()` and `getColour()` methods. Instead, I would much rather have a `RecentActivityIcon`, and give it a 
```csharp
public static RecentActivityIcon(RecentActivityType type);
```
ctor. Inline everything in there, don't bother with splitting methods, just create a full icon from start to end for each type with a `switch` statement. Move the redundant `RelativeSizeAxes`, `Height`, etc. specs all there too, for good measure.

Reply:
Okay! Makes sense to me